training/src/app/chuck-noris/chuck-noris-main/chuck-norris-main.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component } from '@angular/core';
+import { JokeGeneratorService } from "../joke-generator.service";
+import { Observable } from "rxjs";
+import { Joke } from "../joke";
+
+@Component({
+	selector: 'app-chuck-norris-main',
+	templateUrl: './chuck-norris-main.component.html',
+	styleUrls: ['./chuck-norris-main.component.scss']
+})
+
+export class ChuckNorrisMain {
+	public jokeList$: Observable<Joke[]>;
+	public joke$: Observable<Joke>;
+	public changeMode: boolean = false;
+	public searchInput: string;
+
+	constructor(private jokeService: JokeGeneratorService) {
+	}
+
+	public getJokesFromSearch(inputKey: string): void {
+		if (this.searchInput != inputKey) {
+			this.jokeList$ = this.jokeService.getJokeList(inputKey);
+		}
+		this.searchInput = inputKey;
+	}
+
+	onGenerateJokeClick(): void {
+		this.joke$ = this.jokeService.generateJoke();
+	}
+
+	onChangeModeClick(mode: string): void {
```

Comment:
it doesn't work.. if you click twice on random it will change its color every time, while you want that the clicked button will be red not changed every time you click on it